tests/test_spark_cni.py

```diff
@@ -0,0 +1,224 @@
+
+
+@pytest.mark.sanity
+def test_cni(setup_spark):
+    utils.run_tests(app_url=utils.SPARK_EXAMPLES,
+                    app_args="",
+                    expected_output="Pi is roughly 3",
+                    args=["--conf spark.mesos.network.name=dcos",
+                          "--class org.apache.spark.examples.SparkPi"])
+
+
+@pytest.mark.sanity
+@pytest.mark.smoke
+def test_cni_labels(setup_spark):
+    driver_task_id = utils.submit_job(app_url=utils.SPARK_EXAMPLES,
+                                      app_args="3000",   # Long enough to examine the Driver's & Executor's task infos
+                                      args=["--conf spark.mesos.network.name=dcos",
+                                            "--conf spark.mesos.network.labels=key1:val1,key2:val2",
+                                            "--conf spark.cores.max={}".format(CNI_TEST_NUM_EXECUTORS),
+                                            "--class org.apache.spark.examples.SparkPi"])
+
+    # Wait until executors are running
+    sdk_tasks.check_running(SPARK_PI_FW_NAME, CNI_TEST_NUM_EXECUTORS, timeout_seconds=600)
+
+    # Check for network name / labels in Driver task info
+    driver_task = shakedown.get_task(driver_task_id, completed=False)
+    _check_task_network_info(driver_task)
+
+    # Check for network name / labels in Executor task info
+    executor_task = shakedown.get_service_tasks(SPARK_PI_FW_NAME)[0]
+    _check_task_network_info(executor_task)
+
+    # Check job output
+    utils.check_job_output(driver_task_id, "Pi is roughly 3")
+
+
+def _check_task_network_info(task):
+    # Expected: "network_infos":[{
+    #   "name":"dcos",
+    #   "labels":{
+    #       "labels":[
+    #           {"key":"key1","value":"val1"},
+    #           {"key":"key2","value":"val2"}]}}]
+    network_info = task['container']['network_infos'][0]
+    assert network_info['name'] == "dcos"
+    labels = network_info['labels']['labels']
+    assert len(labels) == 2
+    assert labels[0]['key'] == "key1"
+    assert labels[0]['value'] == "val1"
+    assert labels[1]['key'] == "key2"
+    assert labels[1]['value'] == "val2"
+
+
+# The following dispatcher fixtures rely on sdk_install.install because for the time being
+# spark_utils.require_spark can't be effectively used with virtual network due to
+# https://jira.mesosphere.com/browse/DCOS-45468 (Add supporting endpoints for services
+# running in CNI (e.g. Calico) to AdminRouter)
+@pytest.fixture(scope="module")
+def spark_docker_dispatcher(configure_security, configure_universe):
+    sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, CNI_DISPATCHER_SERVICE_NAME, zk=CNI_DISPATCHER_ZK)
+    try:
+        _install_spark_dispatcher(CNI_SERVICE_OPTIONS)
+        yield
+    finally:
+        sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, CNI_DISPATCHER_SERVICE_NAME, zk=CNI_DISPATCHER_ZK)
+
+
+@pytest.fixture(scope="module")
+def spark_ucr_dispatcher(configure_security, configure_universe):
```

Comment:
These two fixtures are near identical. Can you merge and perhaps pass empty options instead for the docker dispatcher (or non-empty for UCR, up to you)?

Reply:
good point, replaced with a parameterized function-scoped fixture